### ISSUE TITLE
Use binding name, env flag instead of namespace id

### ISSUE THREAD

**Issue by ashleymichal** [cloudflare/wrangler#524]
## Existing Design

Currently most kv commands require that you specify `namespace id` as a positional argument:

``` sh
$ wrangler kv:key put 9d28bd2bbb2c4e489006d7d8afa61f07 "key" "value"
```

This is ergonomically awkward and will likely result in a lot of cursing, copy-pasting, and invocations of `wrangler kv:namespace list` or `cat wrangler.toml` to find the appropriate namespace id for a given command.

## New Design

We can instead use the binding name for a given namespace in a project to provide a human-readable/grokable token for identifying a particular namespace in relation to the project. We can also benefit from environments here by allowing for default values (top level, see environments milestone) and accepting an `--env` flag:

``` sh
$ wrangler kv:key put FOO "key" "value" [--env <environment>]
```
This means that given a toml that looks like this:

``` toml
name = "worker-site"
type = "webpack"
account_id = "43434f8d0d9f52a395a1472b35f23439"
workers_dot_dev = true
kv-namespaces = [
	{ binding = "FOO", id = "9d28bd2bbb2c4e489006d7d8afa61f07" },
	{ binding = "BAR", id = "f5b309d7ebec48649ec0604facb26c7d" }
]

[env.staging]
kv-namespaces = [
	{ binding = "FOO", id = "c417513a199c4289a1a0c8385c58a0e7" }
]

[env.prod]
kv-namespaces = [
	{ binding = "FOO", id = "1f44a56679a040f0bb01ba39599ba704" }
]
```

the following command would `put` to the top level binding named "STATIC_CONTENT" (id: 9d28bd2bbb2c4e489006d7d8afa61f07)

``` sh
$ wrangler kv:key put FOO "key" "value"
```

and this would `put` to the binding named "FOO" under env.staging (id: c417513a199c4289a1a0c8385c58a0e7):

``` sh
$ wrangler kv:key put FOO "key" "value" --env=staging
```

### Why not use the "title" of the namespace?

In the context of a Wrangler project, a user is more likely to be interacting with the binding they are using in the worker. There's also not a compelling reason to store namespace titles locally, so using title to refer to a namespace in this way would require an API lookup query on every command invocation, as opposed to the project toml parsing Wrangler already does as part of every command.

### What if you want to work with a namespace you have not added to a project?

We should retain the namespace id argument as a flag and allow a `--` positional argument for the binding name to allow a user to operate on a namespace that is not yet specified in their wrangler project:

``` sh
$ wrangler kv:key put -- "key" "value" --namespace_id=f5b309d7ebec48649ec0604facb26c7d
```

**Comment by gabbifish** [comment]
Blocked until environments support (https://github.com/cloudflare/wrangler/milestone/12) is complete.

**Comment by gabbifish** [comment]
Could I propose an alternative to "What if you want to work with a namespace you have not added to a project"?

Perhaps we can make the syntax clearer/easier by allowing wrangler users to pass in a namespace_id in the first positional argument (much like namespace binding), and allow them to specify an optional `-i/--id` flag for when they want to use a raw namespace id. e.g.
`wrangler kv:namespace rename <namespace_id> <new_name> -i`

**Comment by ashleymichal** [comment]
change positional binding argument to flag `--binding` OR `--namespace-id`

**Comment by gabbifish** [comment]
Closed by #542.
